### PR TITLE
xcdr Test: Remove Workarounds, Commented Out Code

### PR DESCRIPTION
**Pull request by iguessthislldo**
Mostly code having to do with unions that was deferred last time.

### CODE REVIEW

File: tests/DCPS/Compiler/xcdr/xcdr.cpp [hunk 1]
```diff
@@ -2188,198 +2184,225 @@ void set_values<ComplexKeyedStruct, KeyOnly<ComplexKeyedStruct> >(ComplexKeyedSt
...
+  FieldFilter field_filter, bool keyed)
 {
-  key_only_expect_values_equal_base_keys(a, b);
-  EXPECT_EQ(a.extra_value, b.extra_value);
+  if (!(field_filter == FieldFilter_KeyOnly && !keyed)) {
+    EXPECT_EQ(a.long_value, b.long_value);
+    expect_arrays_are_equal(a.long_array_value, b.long_array_value);
+    /* TODO(iguessthislldo): See IDL Def
+    EXPECT_EQ(a.long_seq_value, b.long_seq_value);
+    */
```
Comment (ClaytonCalabrese) [inline tests/DCPS/Compiler/xcdr/xcdr.cpp:2196]: There is still a TODO here, is that expected?

File: tests/DCPS/Compiler/xcdr/xcdr.cpp [hunk 2]
```diff
@@ -2188,198 +2184,225 @@ void set_values<ComplexKeyedStruct, KeyOnly<ComplexKeyedStruct> >(ComplexKeyedSt
...
+    field_filter == FieldFilter_All ? FieldFilter_All : FieldFilter_NestedKeyOnly;
+
+  if (include_possible_keyed) {
+    key_only_tests_struct_expect_values_equal(
+      a.unkeyed_struct_value, b.unkeyed_struct_value, child, false);
+    key_only_tests_struct_expect_values_equal(
+      a.unkeyed_struct_array_value[0], b.unkeyed_struct_array_value[0], child, false);
+    key_only_tests_struct_expect_values_equal(
+      a.unkeyed_struct_array_value[1], b.unkeyed_struct_array_value[1], child, false);
+    /* expect_values_equal(a.unkeyed_struct_seq_value, b.unkeyed_struct_seq_value, child); */
```
Comment (ClaytonCalabrese) [inline tests/DCPS/Compiler/xcdr/xcdr.cpp:2332]: You have a few other commented out lines here, is this also intentional?

File: tests/DCPS/Compiler/xcdr/xcdr.cpp (hunk 1 above)
Comment (iguessthislldo) [reply tests/DCPS/Compiler/xcdr/xcdr.cpp:2196]: Yes, it says what to look at, which points to the Jira ticket.

File: tests/DCPS/Compiler/xcdr/xcdr.cpp [hunk 3]
```diff
@@ -2507,19 +2530,19 @@ void build_expected_complex_struct(DataVec& expected, FieldFilter field_filter,
     // TODO(iguessthislldo): keyed_struct_seq_value would go here
```
Comment (ClaytonCalabrese) [inline tests/DCPS/Compiler/xcdr/xcdr.cpp:2530]: Same question about this TODO

Comment (iguessthislldo) [reply tests/DCPS/Compiler/xcdr/xcdr.cpp:2530]: All of these are the same TODO

Comment (iguessthislldo) [reply tests/DCPS/Compiler/xcdr/xcdr.cpp:2530]: Look at the IDL file

**Review by ClaytonCalabrese — CHANGES_REQUESTED** [review]
If the commented out lines are intentional this is ready to go.

**Review by ClaytonCalabrese — APPROVED** [review]

**Pull request merged**